### ISSUE TITLE
Incorrectly escaping / in build names

### ISSUE THREAD

**Issue by Wilfred** [jenkinsci/stashnotifier-plugin#109]
If I have a Jenkins build whose name is `#7-foo/bar`, then Bitbucket shows `#7-foo\/bar` in its modal popup.

![slash_escape](https://cloud.githubusercontent.com/assets/70800/16194101/bce0da38-36ea-11e6-9718-afa21ac06b7a.png)

This is common when you name your jenkins builds according to the git branches, so `feature/foobar` adds a slash to the build name.

I did file [an issue on bitbucket](https://jira.atlassian.com/browse/BSERV-8792), but it looks like the bug is in the stashnotifier plugin. They've suggested [the escaping logic here](https://github.com/jenkinsci/stashnotifier-plugin/blob/release/1.x/src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java#L779-L784) is causing the problem.

**Comment by scaytrase** [comment]
Yeah, I've seen this before. Could you compose a json body which produces correct message for you build and post it here?

Here are the docs https://developer.atlassian.com/bitbucket/server/docs/latest/how-tos/updating-build-status-for-commits.html

**Comment by Wilfred** [comment]
Sure, here's an example of a cURL request which does the right thing:

```
curl -u testuser:password -H "Content-Type: application/json" -X POST "http://git.internal/rest/build-status/1.0/commits/071c0a5bbe91233b43343c1aa44a374c2dca9e95" -d '{
	"state": "SUCCESSFUL",
	"key": "TEST",
	"name": "helloworld #7-feature/do-stuff",
	"url": "http://jenkins.example.com",
	"description": "built by Jenkins @ http://ci/build/"
}'
```

**Comment by scaytrase** [comment]
@Wilfred isn't this fixed already by
https://github.com/jenkinsci/stashnotifier-plugin/pull/180 ?

**Comment by Wilfred** [comment]
Sounds like it. I don't have a Jenkins instance to test on I'm afraid.
